extensions/cl_exp_tensor.asciidoc

```diff
@@ -0,0 +1,1058 @@
+:data-uri:
+:icons: font
+//include::../config/attribs.txt[]
+//include::{generated}/api/api-dictionary.asciidoc[]
+:source-highlighter: coderay
+
+= cl_exp_tensor
+
+This extension provides new buffer abstraction - tensor objects - for
+managing N-dimensional data.
+
+== XXX - Not complete yet!!!
+
+== Name Strings
+
+`cl_exp_tensor`
+
+== Contact
+
+TODO
+
+== Contributors
+
+Henry Linjamäki, Intel. +
+Pekka Jääslkeläinen, Intel and Tampere University. +
+Ben Ashbaugh, Intel. +
+
+== Notice
+
+TODO
+
+== Status
+
+Draft spec, NOT APPROVED!!
+
+== Version
+
+Built On: {docdate} +
+Version: 0.2.0
+
+== Dependencies
+
+This extension is written against the OpenCL Specification version 3.0.14.
+
+This extension requires OpenCL 1.2 or later.
+
+== Overview
+
+The extension provides new tensor object abstraction. Tensor objects
+are similar to image types in regard they represents N-dimensional
+data of some application chosen data type and they may be mapped to
+dedicated hardware except that
+
+* higher than 3-dimensional data can be supported (limited by
+  devices' capabilities).
+
+* applications may choose how the data elements of the tensors are
+  laid out in the buffers using the tensor layout descriptions
+  provided in this extension.
+
+Applications may also choose the memory layouts of the tensors be
+implementation-specified, letting the driver to optimize the tensor
+data layout for better performance or to lay out the data as required by
+hardware functions (e.g. exposed via builtin kernels).
+
+The scope of this extension to provide host APIs for creating tensor
+objects and transfer data between tensors, host and other memory
+objects.
+
+A separate extension implemented on top of this extension,
+cl_exp_defined_builtin_kernels which provides "defined built-in
+kernels" (DKBs) which operates on tensors. It also provides mechanism
+for drivers to create DBKs that are optimized for the tensor arguments
+they operate on.
+
```

Comment:
Perhaps mention that it's designed for use with command buffers for whole task-graph optimizations. And maybe also about Vulkan layering (if those referred APIs are public).